app/controllers/api/providers_controller.rb

```diff
@@ -50,8 +51,28 @@ def delete_resource(type, id = nil, _data = nil)
       end
     end
 
+    def custom_attributes_edit_resource(object, type, id, data = nil)
+      formatted_data = format_provider_custom_attributes(data)
+      super(object, type, id, formatted_data)
+    end
+
+    def custom_attributes_add_resource(object, type, id, data = nil)
+      formatted_data = format_provider_custom_attributes(data)
+      super(object, type, id, formatted_data)
+    end
+
     private
 
+    def format_provider_custom_attributes(attribute)
+      if CustomAttribute::ALLOWED_API_FIELD_TYPES.include? attribute["field_type"]
+        attribute["value"] = attribute.delete("field_type").safe_constantize.parse(attribute["value"])
+      end
+      attribute["section"] = "metadata" unless @req.action.include?("edit")
+      attribute
+    rescue => err
+      raise BadRequestError, "Could not update the provider custom attributes - #{err}"
```

Comment:
what error are we catching here ? we haven't started to update the attributes yes.

Looks like possible formatting error, so maybe error should be:

     raise BadRequestError, "Invalid provider custom attributes specified - #{err}"